cmd/wallet-sdk-gomobile/display/resolveopts.go

```diff
@@ -10,14 +10,40 @@ import (
 	"errors"
 
 	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/wrapper"
-	"github.com/trustbloc/wallet-sdk/pkg/common"
 
 	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
 
 	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
 	goapicredentialschema "github.com/trustbloc/wallet-sdk/pkg/credentialschema"
 )
 
+// ResolveOpts contains the various parameters for the Resolve function.
+type ResolveOpts struct {
+	VCs             *api.VerifiableCredentialsArray // Required
+	IssuerURI       string                          // Required
+	PreferredLocale string                          // Optional
+	MetricsLogger   api.MetricsLogger               // Optional
+	httpClient      *wrapper.HTTPClient
+}
+
+// NewResolveOpts creates a new ResolveOpts object. This function only takes in required parameters. Optional parameters
+// can be set by setting the fields on the ResolveOpts object that you get back from this function directly.
+func NewResolveOpts(vcs *api.VerifiableCredentialsArray, issuerURI string) *ResolveOpts {
+	return &ResolveOpts{
+		VCs:       vcs,
+		IssuerURI: issuerURI,
+	}
+}
+
+// AddHeaders adds the given HTTP headers to all REST calls made to the issuer during display resolution.
+func (r *ResolveOpts) AddHeaders(headers *api.Headers) {
+	if r.httpClient == nil {
+		r.httpClient = wrapper.NewHTTPClient()
+	}
```

Comment:
This part looks a little bit hacky, better to store headers in ResolveOpts, and create httpClient on config unwrap in resolveDisplayOpts func

Reply:
Done - I also took this opportunity to clean up and simplify the `prepareOpts` function. Take a look and let me know if you have any other suggestions